### PR TITLE
fix #3105 by adding -Werror=format on CI

### LINKED ISSUE #3105: Build failed with -Werror=format
Description:
After commit [cd14742f2](https://github.com/h2o/h2o/commit/cd14742f21c6afc87b76033e76620f86e113f719) h2o can not build

Expected Behavior:
Can build

Output of `make`:
full version is https://gist.github.com/msmhrt/8e1b63d1fb21b85f5c37c4a1553b1018

~~~
...
[  3%] Building C object CMakeFiles/h2o.dir/lib/common/url.c.o
/home/msmhrt/repository/h2o/lib/common/url.c: In function ‘h2o_url_init_with_hostport’:
/home/msmhrt/repository/h2o/lib/common/url.c:430:39: error: ‘%u’ directive writing between 1 and 10 bytes into a region of si
ze 6 [-Werror=format-overflow=]
         int port_len = sprintf(_port, "%" PRIu16, port);
                                       ^~~
/home/msmhrt/repository/h2o/lib/common/url.c:430:40: note: format string is defined here
         int port_len = sprintf(_port, "%" PRIu16, port);
/home/msmhrt/repository/h2o/lib/common/url.c:430:39: note: directive argument in the range [0, 2147483647]
         int port_len = sprintf(_port, "%" PRIu16, port);
                                       ^~~
In file included from /usr/include/stdio.h:862:0,
                 from /home/msmhrt/repository/h2o/include/h2o/memory.h:31,
                 from /home/msmhrt/repository/h2o/lib/common/url.c:26:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:33:10: note: ‘__builtin___sprintf_chk’ output between 2 and 11 bytes into a desti
nation of size 6
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/msmhrt/repository/h2o/lib/common/url.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-nullability-completeness’
cc1: some warnings being treated as errors
CMakeFiles/h2o.dir/build.make:1603: recipe for target 'CMakeFiles/h2o.dir/lib/common/url.c.o' failed
make[2]: *** [CMakeFiles/h2o.dir/lib/common/url.c.o] Error 1
CMakeFiles/Makefile2:883: recipe for target 'CMakeFiles/h2o.dir/all' failed
make[1]: *** [CMakeFiles/h2o.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
msmhrt@ashrose:~/repository/h2o/build$
~~~

### PR DESCRIPTION
**Pull request by gfx**
discussions:

* https://github.com/h2o/h2o/issues/3105#issuecomment-1232336823
* https://github.com/h2o/h2o/pull/3106

### CODE REVIEW

File: misc/docker-ci/check.mk [hunk 1]
```diff
@@ -1,9 +1,9 @@
 CONTAINER_NAME=h2oserver/h2o-ci:ubuntu1604
 SRC_DIR=/h2o
 CHECK_MK=$(SRC_DIR)/misc/docker-ci/check.mk
-CMAKE_ARGS=
-BUILD_ARGS=
-TEST_ENV=
+CMAKE_ARGS?=
```
Comment (kazuho) [inline misc/docker-ci/check.mk:4]: Do we want to add the flags here instead?

I say this because what we can test depends on the images that we control.

Comment (gfx) [reply misc/docker-ci/check.mk]: Ah, yes, we don't have to make things too complicated. Fixed: 01cb9dc8cce712631f5fd5df706aa9391fe0c0fd

File: misc/docker-ci/check.mk [hunk 2]
```diff
@@ -24,37 +25,38 @@ ossl1.1.0+fuzz:
 	docker run $(DOCKER_RUN_OPTS) $(CONTAINER_NAME) \
 		env CC=clang CXX=clang++ \
 		make -f $(SRC_DIR).ro/misc/docker-ci/check.mk _check \
-		CMAKE_ARGS='-DOPENSSL_ROOT_DIR=/opt/openssl-1.1.0 -DBUILD_FUZZER=ON' \
+		CMAKE_ARGS="$(CMAKE_ARGS) -DOPENSSL_ROOT_DIR=/opt/openssl-1.1.0 -DBUILD_FUZZER=ON" \
```
Comment (kazuho) [inline misc/docker-ci/check.mk:28]: Do we need this change and the changes below? 

We are setting the default for each invocation of `check.mk` right above at line 5.

File: misc/docker-ci/check.mk [hunk 3]
```diff
@@ -1,7 +1,8 @@
 CONTAINER_NAME=h2oserver/h2o-ci:ubuntu1604
 SRC_DIR=/h2o
 CHECK_MK=$(SRC_DIR)/misc/docker-ci/check.mk
-CMAKE_ARGS=
+# Specify the compiler flags for CI that might have side effects.
```
Comment (kazuho) [inline misc/docker-ci/check.mk:4]:
```suggestion
# promoting warnings to errors help us detect issues without dealing with false positives generated by every compiler that people might use
```

**Review by kazuho — COMMENTED** [review]
Thank you for the changes.

File: misc/docker-ci/check.mk [hunk 4]
```diff
@@ -1,7 +1,8 @@
 CONTAINER_NAME=h2oserver/h2o-ci:ubuntu1604
 SRC_DIR=/h2o
 CHECK_MK=$(SRC_DIR)/misc/docker-ci/check.mk
-CMAKE_ARGS=
+# promoting warnings to errors help us detect issues without dealing with false positives generated by every compiler that people might use
+CMAKE_ARGS=-Werror=format CMAKE_CXX_FLAGS=-Werror=format
```
Comment (kazuho) [inline misc/docker-ci/check.mk:5]: Shouldn't this be `CMAKE_ARGS=CMAKE_C_FLAGS=-Werror=format ...`?

But moreover ...

File: misc/docker-ci/check.mk [hunk 5]
```diff
@@ -24,37 +25,38 @@ ossl1.1.0+fuzz:
...
+		CMAKE_ARGS="$(CMAKE_ARGS)" \
 		BUILD_ARGS='$(BUILD_ARGS)' \
 		TEST_ENV='$(TEST_ENV)'
 
 dtrace+asan:
 	docker run $(DOCKER_RUN_OPTS) h2oserver/h2o-ci:ubuntu2004 \
 		env DTRACE_TESTS=1 \
 		make -f $(SRC_DIR).ro/misc/docker-ci/check.mk _check \
-		CMAKE_ARGS='-DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_FLAGS=-fsanitize=address -DCMAKE_CXX_FLAGS=-fsanitize=address' \
+		CMAKE_ARGS="$(CMAKE_ARGS) -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_FLAGS=-fsanitize=address -DCMAKE_CXX_FLAGS=-fsanitize=address" \
```
Comment (kazuho) [inline misc/docker-ci/check.mk:51]: This does not work because, we set CMAKE_C_FLAGS after `$(CMAKE_ARGS)`.

**Review by kazuho — COMMENTED** [review]
Rechecking this PR, removal of `-Werror=format` is what we've agreed upon. However, the changes to CI to use `-Werror=format` seems to be corrupt.

I will propose a different PR.

**Pull request closed without merging**